// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/DateFilter/components/DayRangeInput/DayRangeInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DayRangeInput = void 0;

var _components = require("@looker/components");

var _react = _interopRequireDefault(require("react"));

var _format_date = require("../../utils/format_date");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const DayRangeInput = ({
  value,
  onChange
}) => {
  const handleChange = (d = {}) => {
    const newFrom = d.from || new Date(Date.now());
    const newTo = d.to || new Date(Date.now());
    const newDateRange = {
      from: new Date(newFrom.getFullYear(), newFrom.getMonth(), newFrom.getDate()),
      to: new Date(newTo.getFullYear(), newTo.getMonth(), newTo.getDate())
    };
    onChange(newDateRange);
  };

  const formattedValue = `${(0, _format_date.formatDate)(value.from)} \u2013 ${(0, _format_date.formatDate)(value.to)}`;
  return _react.default.createElement(_components.Popover, {
    content: _react.default.createElement(_components.Box2, {
      p: "u3"
    }, _react.default.createElement(_components.InputDateRange, {
      onChange: handleChange,
      value: value,
      dateStringFormat: _format_date.FILTERS_DATE_FORMAT
    })),
    placement: "bottom-start"
  }, _react.default.createElement(_components.ChipButton, null, formattedValue));
};

exports.DayRangeInput = DayRangeInput;
