// @looker/filter-components/Filter/components/AdvancedFilter/components/UserAttributes/UserAttributes.js
import "core-js/modules/es.array.map.js";
import "core-js/modules/es.function.name.js";
import "core-js/modules/es.array.concat.js";
import { findUserAttribute } from '@looker/filter-expressions';
import React from 'react';
import { GroupSelect } from '../GroupSelect';

var createOptions = function createOptions() {
  var userAttributes = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : [];
  return userAttributes.map(function (_ref) {
    var name = _ref.name,
        label = _ref.label,
        value = _ref.value;
    return {
      value: name,
      label: "".concat(label, " (").concat(value, ")")
    };
  });
};

export var UserAttributes = function UserAttributes(_ref2) {
  var item = _ref2.item,
      attributeName = _ref2.item.attributeName,
      userAttributes = _ref2.userAttributes,
      onChange = _ref2.onChange,
      validationMessage = _ref2.validationMessage;

  var userAttributeChange = function userAttributeChange(value) {
    var userAttribute = findUserAttribute(value, userAttributes);
    onChange(item.id, {
      attributeName: value,
      attributeValue: userAttribute && userAttribute.value
    });
  };

  return React.createElement(GroupSelect, {
    value: attributeName,
    placeholder: "Select...",
    options: createOptions(userAttributes),
    onChange: userAttributeChange,
    validationType: validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type,
    placement: "right"
  });
};
