// @looker/filter-components/Filter/constants/date/before_after_units.js
import "core-js/modules/es.array.concat.js";
import "core-js/modules/es.array.is-array.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.symbol.description.js";
import "core-js/modules/es.object.to-string.js";
import "core-js/modules/es.symbol.iterator.js";
import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.string.iterator.js";
import "core-js/modules/web.dom-collections.iterator.js";
import "core-js/modules/es.array.from.js";
import "core-js/modules/es.array.slice.js";
import "core-js/modules/es.function.name.js";

function _toConsumableArray(arr) { return _arrayWithoutHoles(arr) || _iterableToArray(arr) || _unsupportedIterableToArray(arr) || _nonIterableSpread(); }

function _nonIterableSpread() { throw new TypeError("Invalid attempt to spread non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _iterableToArray(iter) { if (typeof Symbol !== "undefined" && iter[Symbol.iterator] != null || iter["@@iterator"] != null) return Array.from(iter); }

function _arrayWithoutHoles(arr) { if (Array.isArray(arr)) return _arrayLikeToArray(arr); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

import { useTranslation } from '../../../utils';

var useAgoDateUnits = function useAgoDateUnits() {
  var _useTranslation = useTranslation('before_after_units'),
      t = _useTranslation.t;

  return [{
    value: 'year',
    unit: 'years',
    label: t('years ago')
  }, {
    value: 'quarter',
    unit: 'quarters',
    label: t('quarters ago')
  }, {
    value: 'month',
    unit: 'months',
    label: t('months ago')
  }, {
    value: 'week',
    unit: 'weeks',
    label: t('weeks ago')
  }, {
    value: 'day',
    unit: 'days',
    label: t('days ago')
  }, {
    value: 'hour',
    unit: 'hours',
    label: t('hours ago')
  }, {
    value: 'minute',
    unit: 'minutes',
    label: t('minutes ago')
  }, {
    value: 'second',
    unit: 'seconds',
    label: t('seconds ago')
  }, {
    value: 'now',
    unit: 'now',
    label: t('now')
  }];
};

var useFromNowDateUnits = function useFromNowDateUnits() {
  var _useTranslation2 = useTranslation('before_after_units'),
      t = _useTranslation2.t;

  return [{
    value: 'f_second',
    unit: 'second',
    label: t('seconds from now'),
    fromnow: true
  }, {
    value: 'f_minute',
    unit: 'minute',
    label: t('minutes from now'),
    fromnow: true
  }, {
    value: 'f_hour',
    unit: 'hour',
    label: t('hours from now'),
    fromnow: true
  }, {
    value: 'f_day',
    unit: 'day',
    label: t('days from now'),
    fromnow: true
  }, {
    value: 'f_week',
    unit: 'week',
    label: t('weeks from now'),
    fromnow: true
  }, {
    value: 'f_month',
    unit: 'month',
    label: t('months from now'),
    fromnow: true
  }, {
    value: 'f_quarter',
    unit: 'quarter',
    label: t('quarters from now'),
    fromnow: true
  }, {
    value: 'f_year',
    unit: 'year',
    label: t('years from now'),
    fromnow: true
  }];
};

export var useBeforeOrAfterUnits = function useBeforeOrAfterUnits() {
  var agoDateUnits = useAgoDateUnits();
  var fromNowDateUnits = useFromNowDateUnits();
  return [].concat(_toConsumableArray(agoDateUnits), _toConsumableArray(fromNowDateUnits));
};
export var useFiscalBeforeOrAfterUnits = function useFiscalBeforeOrAfterUnits() {
  var _useTranslation3 = useTranslation('before_after_units'),
      t = _useTranslation3.t;

  var agoDateUnits = useAgoDateUnits();
  var fromNowDateUnits = useFromNowDateUnits();
  return [{
    value: 'fiscal year',
    unit: 'fiscal year',
    label: t('fiscal years ago')
  }, {
    value: 'fiscal quarter',
    unit: 'fiscal quarter',
    label: t('fiscal quarters ago')
  }].concat(_toConsumableArray(agoDateUnits), _toConsumableArray(fromNowDateUnits), [{
    value: 'f_fiscal quarter',
    unit: 'fiscal quarter',
    label: t('fiscal quarter from now')
  }, {
    value: 'f_fiscal year',
    unit: 'fiscal year',
    label: t('fiscal years from now')
  }]);
};
