// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/LocationFilter/components/LocationBox/LocationBox.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LocationBox = void 0;

var _components = require("@looker/components");

var _pick = _interopRequireDefault(require("lodash/pick"));

var _react = _interopRequireDefault(require("react"));

var _LocationExact = require("../LocationExact");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const LocationBox = ({
  item,
  onChange
}) => _react.default.createElement(_components.Flex, {
  flexDirection: "row"
}, _react.default.createElement(_LocationExact.LocationExact, {
  item: (0, _pick.default)(item, ['id', 'is', 'lat', 'lon', 'type']),
  onChange: onChange,
  placement: "middle",
  latString: "FROM LATITUDE"
}), _react.default.createElement(_LocationExact.LocationExact, {
  lat: "lat1",
  lon: "lon1",
  item: {
    id: item.id,
    is: item.is,
    lat: item.lat1,
    lon: item.lon1,
    type: item.type
  },
  onChange: onChange,
  latString: "TO LATITUDE"
}));

exports.LocationBox = LocationBox;
