// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/UserAttributes/UserAttributes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UserAttributes = void 0;

require("core-js/modules/es.array.map.js");

var _filterExpressions = require("@looker/filter-expressions");

var _react = _interopRequireDefault(require("react"));

var _GroupSelect = require("../GroupSelect");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const createOptions = (userAttributes = []) => userAttributes.map(({
  name,
  label,
  value
}) => ({
  value: name,
  label: `${label} (${value})`
}));

const UserAttributes = ({
  item,
  item: {
    attributeName
  },
  userAttributes,
  onChange,
  validationMessage
}) => {
  const userAttributeChange = value => {
    const userAttribute = (0, _filterExpressions.findUserAttribute)(value, userAttributes);
    onChange(item.id, {
      attributeName: value,
      attributeValue: userAttribute && userAttribute.value
    });
  };

  return _react.default.createElement(_GroupSelect.GroupSelect, {
    value: attributeName,
    placeholder: "Select...",
    options: createOptions(userAttributes),
    onChange: userAttributeChange,
    validationType: validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type,
    placement: "right"
  });
};

exports.UserAttributes = UserAttributes;
