// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/TierFilter/TierFilter.js
"use strict";

require("core-js/modules/es.array.index-of.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TierFilter = void 0;

require("core-js/modules/es.array.map.js");

require("core-js/modules/es.array.includes.js");

require("core-js/modules/es.string.includes.js");

var _filterExpressions = require("@looker/filter-expressions");

var _react = _interopRequireDefault(require("react"));

var _GroupSelect = require("../GroupSelect");

var _tier_filter_type_to_filter = require("./utils/tier_filter_type_to_filter");

var _utils = require("../../utils");

var _ItemLayout = require("../ItemLayout");

var _utils2 = require("../../../../utils");

const _excluded = ["item", "filterType", "enumerations", "field", "userAttributes", "onChange", "showMatchesAdvanced", "validationMessage"];

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

const TierFilter = _ref => {
  let item = _ref.item,
      filterType = _ref.filterType,
      enumerations = _ref.enumerations,
      field = _ref.field,
      userAttributes = _ref.userAttributes,
      onChange = _ref.onChange,
      showMatchesAdvanced = _ref.showMatchesAdvanced,
      validationMessage = _ref.validationMessage,
      rest = _objectWithoutProperties(_ref, _excluded);

  const typeChange = value => onChange(item.id, (0, _filterExpressions.convertOptionToType)(String(value)));

  const selectValue = (0, _filterExpressions.convertTypeToOption)(item);
  const isParamFilter = field ? field.category === 'parameter' : false;
  const unescapeEnumerationValues = (field === null || field === void 0 ? void 0 : field.has_allowed_values) && isParamFilter;
  const unescapedEnumerations = enumerations === null || enumerations === void 0 ? void 0 : enumerations.map((0, _utils2.createEnumeration)(unescapeEnumerationValues));
  const FilterComponent = (0, _tier_filter_type_to_filter.tierFilterTypeToFilter)(item.type, isParamFilter, rest.allowMultipleOptions);
  const isValueSet = item.value && item.value.length > 0;
  const isValueInEnumeration = unescapedEnumerations === null || unescapedEnumerations === void 0 ? void 0 : unescapedEnumerations.some(e => {
    var _item$value;

    return (_item$value = item.value) === null || _item$value === void 0 ? void 0 : _item$value.includes(e.value);
  });

  if (isParamFilter && unescapedEnumerations !== null && unescapedEnumerations !== void 0 && unescapedEnumerations.length && isValueSet && !isValueInEnumeration) {
    item.value = [String(unescapedEnumerations[0].value)];
  }

  const paramFilterOptions = (0, _utils.useParamFilterOptions)();
  const tierFilterOptions = (0, _utils.useTierFilterOptions)();
  const options = (0, _utils.useFilterOptions)(isParamFilter ? paramFilterOptions : tierFilterOptions, isParamFilter ? false : showMatchesAdvanced);
  return _react.default.createElement(_ItemLayout.ItemLayout, _extends({
    item: item
  }, rest), _react.default.createElement(_GroupSelect.GroupSelect, {
    value: selectValue,
    options: options,
    onChange: typeChange,
    validationType: validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type,
    placement: item.type === 'anyvalue' ? undefined : 'left'
  }), _react.default.createElement(FilterComponent, {
    item: item,
    onChange: onChange,
    borderRadiusLeft: 0,
    disableCreate: true,
    validationMessage: validationMessage,
    enumerations: unescapedEnumerations,
    userAttributes: userAttributes,
    filterType: filterType,
    placement: "right"
  }));
};

exports.TierFilter = TierFilter;
