// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/NumberFilter/components/Between/Between.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Between = void 0;

var _components = require("@looker/components");

var _react = _interopRequireDefault(require("react"));

var _utils = require("../../../../../../../utils");

var _GroupInput = require("../../../GroupInput");

var _GroupSelect = require("../../../GroupSelect");

var _MidInputLabel = require("../../../MidInputLabel");

var _utils2 = require("../../../../utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const Between = ({
  item,
  onChange,
  validationMessage
}) => {
  const _useTranslation = (0, _utils.useTranslation)('Between'),
        t = _useTranslation.t;

  const betweenOptions = (0, _utils2.useBetweenOptions)();
  const _item$high = item.high,
        high = _item$high === void 0 ? '' : _item$high,
        _item$low = item.low,
        low = _item$low === void 0 ? '' : _item$low,
        id = item.id;

  const lowChange = event => {
    onChange === null || onChange === void 0 ? void 0 : onChange(id, {
      low: event.target.value
    });
  };

  const highChange = event => {
    onChange === null || onChange === void 0 ? void 0 : onChange(id, {
      high: event.target.value
    });
  };

  const selectChange = value => {
    onChange === null || onChange === void 0 ? void 0 : onChange(id, {
      bounds: value
    });
  };

  const validationProps = {
    noErrorIcon: true,
    validationType: validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type
  };
  return _react.default.createElement(_components.Space, {
    gap: "none",
    width: "auto"
  }, _react.default.createElement(_GroupSelect.GroupSelect, _extends({
    placement: "middle",
    value: item.bounds,
    options: betweenOptions,
    onChange: selectChange
  }, validationProps)), _react.default.createElement(_GroupInput.GroupInput, _extends({
    value: low,
    type: "number",
    onChange: lowChange,
    minWidth: "4.5em",
    "data-testid": "low"
  }, validationProps)), _react.default.createElement(_MidInputLabel.MidInputLabel, {
    validationType: validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type
  }, t('AND')), _react.default.createElement(_GroupInput.GroupInput, _extends({
    value: high,
    type: "number",
    onChange: highChange,
    placement: "right",
    minWidth: "4.5em",
    "data-testid": "high"
  }, validationProps)));
};

exports.Between = Between;
