// @looker/filter-components/Filter/components/AdvancedFilter/components/ItemLayout/ItemLayout.js
import { Box2 } from '@looker/components';
import React from 'react';
import { AddRemoveButtons } from '../AddRemoveButtons';
import { OperatorLabel } from '../OperatorLabel';
export var ItemLayout = function ItemLayout(_ref) {
  var children = _ref.children,
      item = _ref.item,
      onAdd = _ref.onAdd,
      onRemove = _ref.onRemove,
      showOperator = _ref.showOperator,
      showAdd = _ref.showAdd,
      showRemove = _ref.showRemove;

  var handleAdd = function handleAdd() {
    return onAdd(item);
  };

  var handleRemove = function handleRemove() {
    return onRemove(item.id);
  };

  return React.createElement(Box2, {
    role: "group",
    display: "flex",
    alignItems: "center"
  }, showOperator && React.createElement(Box2, {
    width: 44,
    mr: "medium",
    textAlign: "right"
  }, showOperator === true && React.createElement(OperatorLabel, {
    value: item.is
  })), children, React.createElement(AddRemoveButtons, {
    showAdd: showAdd,
    showRemove: showRemove,
    onAdd: handleAdd,
    onRemove: handleRemove
  }));
};
