// @looker/visualizations-table/src/hooks/useHeadlessTable.ts
/*

 MIT License

 Copyright (c) 2022 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */
import { useState, useCallback } from 'react'
import type { SDKRecord, CTable, Fields } from '@looker/visualizations-adapters'
import type { ColumnDef, SortingState, Header } from '@tanstack/table-core'
import {
  getCoreRowModel,
  useReactTable,
  getSortedRowModel,
} from '@tanstack/react-table'
import noop from 'lodash/noop'

type UseHeadlessTableProps = {
  data: SDKRecord[]
  config: CTable
  fields: Fields
}

/**
 * useHeadlessTable handles the configuration of our headless table library based for handling
 * basic table state and sorting logic.
 */

export const useHeadlessTable = ({
  data,
  config,
  fields,
}: UseHeadlessTableProps) => {
  const { column_order } = config

  const flatFields = [...fields.dimensions, ...fields.measures]

  const columns: ColumnDef<SDKRecord>[] = flatFields.map(field => {
    return {
      header: field.label_short,
      accessorFn: data => data[field.name],
      id: field.name,
    }
  })

  const [sorting, setSorting] = useState<SortingState>(
    flatFields
      .map(field => {
        if (field.sorted) {
          return { id: field.name, desc: field.sorted.desc }
        }
        return undefined
      })
      .filter(Boolean) as SortingState
  )

  const handleTableSort = useCallback(
    (header: Header<SDKRecord, unknown>) => {
      if (header?.column?.getCanSort()) {
        const currentDescending = !!sorting.find(
          column => column.id === header.id
        )?.desc
        setSorting([{ id: header.id, desc: !currentDescending }])
      }
    },
    [sorting]
  )

  const table = useReactTable({
    data,
    columns,
    getCoreRowModel: getCoreRowModel(),
    getSortedRowModel: getSortedRowModel(),
    state: {
      columnOrder: column_order,
      columnPinning: { left: [], right: [] },
      sorting,
    },
    onStateChange: noop,
    renderFallbackValue: null,
  })

  return { table, sorting, handleTableSort }
}
