// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/StringFilter/utils/string_filter_type_to_filter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.stringFilterTypeToFilter = void 0;

var _defaultTo = _interopRequireDefault(require("lodash/defaultTo"));

var _MatchesAdvanced = require("../../MatchesAdvanced");

var _UserAttributes = require("../../UserAttributes");

var _MultiStringInput = require("../components/MultiStringInput");

var _StringInput = require("../components/StringInput");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const Blank = () => '';

const filterTypeToStringMap = {
  null: Blank,
  contains: _MultiStringInput.MultiStringInput,
  match: _MultiStringInput.MultiStringInput,
  startsWith: _MultiStringInput.MultiStringInput,
  endsWith: _MultiStringInput.MultiStringInput,
  blank: Blank,
  user_attribute: _UserAttributes.UserAttributes
};
const filterTypeToStringMapSingleValue = {
  contains: _StringInput.StringInput,
  match: _StringInput.StringInput,
  startsWith: _StringInput.StringInput,
  endsWith: _StringInput.StringInput
};

const stringFilterTypeToFilter = (type, isParameterField, allowMultipleValues) => {
  if ((!allowMultipleValues || isParameterField) && filterTypeToStringMapSingleValue[type]) {
    return filterTypeToStringMapSingleValue[type];
  }

  return (0, _defaultTo.default)(filterTypeToStringMap[type], _MatchesAdvanced.MatchesAdvanced);
};

exports.stringFilterTypeToFilter = stringFilterTypeToFilter;
