// @looker/filter-components/Filter/components/AdvancedFilter/components/DateFilter/components/DateInput/DateInput.js
import "core-js/modules/es.date.to-string.js";
import "core-js/modules/es.date.now.js";
import { Box, ChipButton, InputDate, Popover } from '@looker/components';
import React from 'react';
import styled from 'styled-components';
import { inputPlacementStyle } from '../../../../../../utils/filter_styles';
import { FILTERS_DATE_FORMAT, formatDate } from '../../utils/format_date';
export var DateInputInternal = function DateInputInternal(_ref) {
  var className = _ref.className,
      date = _ref.date,
      onChange = _ref.onChange;

  var handleDayChange = function handleDayChange(d) {
    var newDate = d || new Date(Date.now());
    onChange(new Date(newDate.getFullYear(), newDate.getMonth(), newDate.getDate(), date.getHours(), date.getMinutes(), date.getSeconds()));
  };

  return React.createElement(Popover, {
    content: React.createElement(Box, {
      p: "small"
    }, React.createElement(InputDate, {
      defaultValue: date,
      onChange: handleDayChange,
      dateStringFormat: FILTERS_DATE_FORMAT
    })),
    placement: "bottom-start"
  }, React.createElement(ChipButton, {
    className: className
  }, formatDate(date)));
};
export var DateInput = styled(DateInputInternal).withConfig({
  displayName: "DateInput",
  componentId: "sc-iggigb-0"
})(["", ""], inputPlacementStyle);
