// @looker/filter-components/Filter/components/AdvancedFilter/components/DateFilter/components/ThisNextLast/ThisNextLast.js
import React from 'react';
import { useFiscalThisNextUnits, useFiscalLastUnits, useThisNextUnits, useLastUnits } from '../../../../../../constants';
import { showFiscalUnits } from '../../../../utils/show_fiscal_units';
import { GroupSelect } from '../../../GroupSelect';

var useOptions = function useOptions(_ref, field) {
  var type = _ref.type;
  var showFiscal = showFiscalUnits(field);
  var lastUnits = useLastUnits();
  var fiscalLastUnits = useFiscalLastUnits();
  var thisNextUnits = useThisNextUnits();
  var fiscalThisNextUnits = useFiscalThisNextUnits();

  if (type === 'last') {
    return showFiscal ? fiscalLastUnits : lastUnits;
  }

  return showFiscal ? fiscalThisNextUnits : thisNextUnits;
};

export var ThisNextLast = function ThisNextLast(_ref2) {
  var item = _ref2.item,
      onChange = _ref2.onChange,
      field = _ref2.field;
  var id = item.id,
      unit = item.unit;

  var unitChange = function unitChange(value) {
    return onChange(id, {
      unit: value
    });
  };

  var options = useOptions(item, field);
  return React.createElement(GroupSelect, {
    value: unit,
    options: options,
    onChange: unitChange,
    placement: "right"
  });
};
