// @looker/filter-components/cjs/Filter/components/AdvancedFilter/AdvancedFilter.js
"use strict";

require("core-js/modules/es.array.filter.js");

require("core-js/modules/es.object.get-own-property-descriptors.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AdvancedFilter = void 0;

require("core-js/modules/es.array.map.js");

var _react = _interopRequireDefault(require("react"));

var _filterExpressions = require("@looker/filter-expressions");

var _type_to_component = require("../../utils/type_to_component");

var _components = require("@looker/components");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

const getShowOperator = (length, index) => {
  if (index > 0) return true;
  if (length > 1) return 'spacer';
  return false;
};

const AdvancedFilter = ({
  ast,
  updateAST,
  name,
  onInputChange,
  changeFilter,
  suggestions,
  enumerations,
  isLinked,
  expressionType,
  userAttributes,
  field,
  inline,
  validationMessage,
  isLoading,
  allowMultipleValues
}) => {
  const onAdd = (filterItem, keepValue) => {
    if (ast) {
      const newItem = keepValue ? filterItem : _objectSpread(_objectSpread({}, filterItem), {}, {
        value: []
      });
      updateAST((0, _filterExpressions.addNode)(ast, newItem));
    }
  };

  const onRemove = id => {
    if (ast) {
      updateAST((0, _filterExpressions.removeNode)(ast, id));
    }
  };

  const FilterComponent = (0, _type_to_component.typeToComponent)(expressionType);
  if (!FilterComponent) return null;
  const items = (0, _filterExpressions.treeToList)(ast);
  const lastItemIndex = items.length - 1;
  const filterList = items.map((item, itemIndex) => {
    const key = `${name}-${item.id}`;
    const isMatchesAdvanced = item.type === 'matchesAdvanced';
    const showAdd = itemIndex === lastItemIndex && !isMatchesAdvanced && !(field !== null && field !== void 0 && field.parameter) && Boolean(allowMultipleValues);
    const showRemove = lastItemIndex > 0 && !isMatchesAdvanced;
    return _react.default.createElement(FilterComponent, {
      key: key,
      name: name,
      filterType: expressionType,
      isLinked: isLinked,
      suggestions: suggestions,
      enumerations: enumerations,
      item: item,
      isLoading: isLoading,
      onChange: changeFilter,
      onInputChange: onInputChange,
      onAdd: onAdd,
      onRemove: onRemove,
      showAdd: showAdd,
      showName: itemIndex === 0,
      showRemove: showRemove,
      showOperator: getShowOperator(items.length, itemIndex),
      userAttributes: userAttributes,
      showMatchesAdvanced: items.length === 1,
      field: field,
      inline: inline,
      validationMessage: validationMessage,
      allowMultipleOptions: Boolean(allowMultipleValues)
    });
  });
  return _react.default.createElement(_components.SpaceVertical, null, filterList);
};

exports.AdvancedFilter = AdvancedFilter;
