// @looker/filter-components/Filter/components/AdvancedFilter/components/DateFilter/DateFilter.js
var _excluded = ["item", "filterType", "onChange", "onAdd", "showAdd", "showMatchesAdvanced", "validationMessage", "userAttributes", "field"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import "core-js/modules/es.array.includes.js";
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.object.define-property.js";
import "core-js/modules/es.array.filter.js";
import "core-js/modules/es.object.get-own-property-descriptor.js";
import "core-js/modules/es.array.for-each.js";
import "core-js/modules/web.dom-collections.for-each.js";
import "core-js/modules/es.object.get-own-property-descriptors.js";
import "core-js/modules/es.object.define-properties.js";
import "core-js/modules/es.object.assign.js";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import { convertTypeToMatchesAdvancedOption, isDateTime, sanitizeDate } from '@looker/filter-expressions';
import React from 'react';
import { useDateFilterOptions } from '../../utils/get_date_filter_options';
import { GroupSelect } from '../GroupSelect';
import { dateFilterTypeToFilter } from './utils/date_filter_type_to_filter';
import { newDateItem } from './utils/new_date_item';
import { useFilterOptions } from '../../utils';
import { ItemLayout } from '../ItemLayout';
export var DateFilter = function DateFilter(_ref) {
  var item = _ref.item,
      filterType = _ref.filterType,
      onChange = _ref.onChange,
      onAdd = _ref.onAdd,
      showAdd = _ref.showAdd,
      showMatchesAdvanced = _ref.showMatchesAdvanced,
      validationMessage = _ref.validationMessage,
      userAttributes = _ref.userAttributes,
      field = _ref.field,
      rest = _objectWithoutProperties(_ref, _excluded);

  var typeChange = function typeChange(value) {
    return onChange(item.id, sanitizeDate(_objectSpread(_objectSpread({}, item), {}, {
      type: value
    })));
  };

  var isParameter = !!(field !== null && field !== void 0 && field.parameter);
  var dateFilterOptions = useDateFilterOptions(isParameter);
  var type = convertTypeToMatchesAdvancedOption(item);

  if (type === 'matchesAdvanced') {
    showMatchesAdvanced = true;
    showAdd = false;
  }

  var options = useFilterOptions(dateFilterOptions, !isParameter && showMatchesAdvanced);

  var handleOnAdd = function handleOnAdd() {
    return onAdd(newDateItem(item), true);
  };

  var FilterComponent = dateFilterTypeToFilter(item.type);
  return React.createElement(ItemLayout, _extends({
    item: item,
    showAdd: showAdd,
    onAdd: handleOnAdd
  }, rest), React.createElement(GroupSelect, {
    value: type,
    options: options,
    onChange: typeChange,
    validationType: validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type,
    placement: ['null', 'notnull', 'anyvalue'].includes(item.type) ? undefined : 'left'
  }), React.createElement(FilterComponent, {
    item: item,
    onChange: onChange,
    borderRadiusLeft: 0,
    userAttributes: userAttributes,
    showTime: isDateTime(filterType),
    validationMessage: validationMessage,
    filterType: filterType,
    field: field,
    placement: "right"
  }));
};
