// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/NumberFilter/components/SingleNumberInput/SingleNumberInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SingleNumberInput = void 0;

var _react = _interopRequireDefault(require("react"));

var _GroupInput = require("../../../GroupInput");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const SingleNumberInput = ({
  item,
  onChange,
  placement,
  validationMessage
}) => {
  var _ref;

  const inputChange = ({
    currentTarget: {
      value
    }
  }) => {
    const numberValue = Number(value);
    const newValueArr = value === '' ? [] : [numberValue];

    if (!Number.isNaN(numberValue)) {
      onChange === null || onChange === void 0 ? void 0 : onChange(item.id, {
        value: newValueArr
      });
    }
  };

  const inputValue = (_ref = Array.isArray(item.value) ? item.value[0] : item.value) !== null && _ref !== void 0 ? _ref : '';
  return _react.default.createElement(_GroupInput.GroupInput, {
    type: "number",
    value: inputValue,
    onChange: inputChange,
    placement: placement,
    minWidth: "4.5em",
    "data-testid": "single-number",
    validationType: validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type,
    noErrorIcon: true
  });
};

exports.SingleNumberInput = SingleNumberInput;
