// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/DateFilter/components/DateInput/DateInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DateInput = exports.DateInputInternal = void 0;

var _components = require("@looker/components");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _filter_styles = require("../../../../../../utils/filter_styles");

var _format_date = require("../../utils/format_date");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const DateInputInternal = ({
  className,
  date,
  onChange
}) => {
  const handleDayChange = d => {
    const newDate = d || new Date(Date.now());
    onChange(new Date(newDate.getFullYear(), newDate.getMonth(), newDate.getDate(), date.getHours(), date.getMinutes(), date.getSeconds()));
  };

  return _react.default.createElement(_components.Popover, {
    content: _react.default.createElement(_components.Box, {
      p: "small"
    }, _react.default.createElement(_components.InputDate, {
      defaultValue: date,
      onChange: handleDayChange,
      dateStringFormat: _format_date.FILTERS_DATE_FORMAT
    })),
    placement: "bottom-start"
  }, _react.default.createElement(_components.ChipButton, {
    className: className
  }, (0, _format_date.formatDate)(date)));
};

exports.DateInputInternal = DateInputInternal;
const DateInput = (0, _styledComponents.default)(DateInputInternal).withConfig({
  displayName: "DateInput",
  componentId: "sc-iggigb-0"
})(["", ""], _filter_styles.inputPlacementStyle);
exports.DateInput = DateInput;
