// @looker/filter-components/Filter/components/AdvancedFilter/components/DateFilter/components/RelativeTimeframes/components/RelativeTimeframePresets/RelativeTimeframePresets.js
import "core-js/modules/es.array.map.js";
import "core-js/modules/es.object.values.js";
import "core-js/modules/es.array.is-array.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.symbol.description.js";
import "core-js/modules/es.object.to-string.js";
import "core-js/modules/es.symbol.iterator.js";
import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.string.iterator.js";
import "core-js/modules/web.dom-collections.iterator.js";
import "core-js/modules/es.array.slice.js";
import "core-js/modules/es.function.name.js";
import "core-js/modules/es.array.from.js";
import "core-js/modules/es.object.assign.js";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

import { FadeIn, Icon, MenuDivider, MenuItem, MenuList, DialogContext } from '@looker/components';
import { Check } from '@styled-icons/material/Check';
import { ExpandMore } from '@styled-icons/material-rounded/ExpandMore';
import React, { useContext, useState } from 'react';
import styled from 'styled-components';
import { useTranslation } from '../../../../../../../../../utils';
import { useRelativeTimeframePresets } from '../../utils/get_relative_timeframe_presets';
import { PresetTimeframes, PresetTimeframesLeastRecent, PresetTimeframesMostRecent, PresetTimeframesPrevious, PresetTimeframesRecent, PresetTimeframesThis } from '../../../../types/relative_timeframe_types';
export var RelativeTimeframePresets = function RelativeTimeframePresets(props) {
  var _useTranslation = useTranslation('RelativeTimeframePresets'),
      t = _useTranslation.t;

  var _useState = useState(false),
      _useState2 = _slicedToArray(_useState, 2),
      showMore = _useState2[0],
      setShowMore = _useState2[1];

  var handleClick = function handleClick(e) {
    setShowMore(true);
    e.preventDefault();
  };

  return React.createElement(MenuList, {
    density: -1,
    iconGutter: true
  }, showMore ? React.createElement(React.Fragment, null, React.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: PresetTimeframesMostRecent,
    delay: "rapid"
  })), React.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: PresetTimeframesRecent,
    delay: "simple"
  })), React.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: PresetTimeframesLeastRecent,
    delay: "moderate"
  })), React.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: PresetTimeframesThis,
    delay: "complex"
  })), React.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: PresetTimeframesPrevious,
    delay: "intricate"
  }))) : React.createElement(React.Fragment, null, React.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: PresetTimeframes,
    delay: "none",
    duration: "none"
  })), React.createElement(CustomMenuItem, {
    detail: React.createElement(Icon, {
      size: 28,
      color: "ui3",
      pr: "xsmall",
      icon: React.createElement(ExpandMore, null)
    }),
    onClick: handleClick
  }, t('More'))));
};

var PresetTimeframeGroup = function PresetTimeframeGroup(_ref) {
  var _ref$duration = _ref.duration,
      duration = _ref$duration === void 0 ? 'simple' : _ref$duration,
      delay = _ref.delay,
      onPresetChange = _ref.onPresetChange,
      presetTimeframe = _ref.presetTimeframe,
      value = _ref.value;
  var presets = useRelativeTimeframePresets();

  var _useContext = useContext(DialogContext),
      closeModal = _useContext.closeModal;

  var handleOnClick = function handleOnClick(timeframe) {
    return function () {
      onPresetChange(timeframe);
      closeModal === null || closeModal === void 0 ? void 0 : closeModal();
    };
  };

  return React.createElement(React.Fragment, null, Object.values(presetTimeframe).map(function (timeframe, index) {
    var current = typeof value === 'string' && value === timeframe;
    return React.createElement(CustomMenuItem, {
      icon: current ? React.createElement(Check, null) : undefined,
      key: index,
      onClick: handleOnClick(timeframe),
      selected: current
    }, React.createElement(FadeIn, {
      duration: duration,
      delay: delay
    }, presets[timeframe]));
  }), React.createElement(MenuDivider, null));
};

var CustomMenuItem = styled(MenuItem).withConfig({
  displayName: "RelativeTimeframePresets__CustomMenuItem",
  componentId: "sc-1lkgww6-0"
})(["& > button{&[aria-current='true']{background-color:", ";", "{color:", ";}}&:hover{background-color:", ";}}"], function (_ref2) {
  var colors = _ref2.theme.colors;
  return colors.background;
}, Icon, function (_ref3) {
  var colors = _ref3.theme.colors;
  return colors.key;
}, function (_ref4) {
  var colors = _ref4.theme.colors;
  return colors.keySubtle;
});
