// @looker/filter-components/Filter/components/ControlFilter/components/TagList/TagList.js
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.object.assign.js";
var _excluded = ["value", "options", "onChange", "onInputChange", "validationMessage"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import { InputText, SelectMulti } from '@looker/components';
import React, { useEffect, useRef } from 'react';
import styled from 'styled-components';
import { tokenStylePlaceholder } from '../../../../utils/filter_styles';
import { useOptionFiltering } from '../../../../utils/use_option_filtering';
import { usePlaceholder } from '../../../../utils/use_placeholder';

var TagListComponent = function TagListComponent(_ref) {
  var value = _ref.value,
      options = _ref.options,
      onChange = _ref.onChange,
      onInputChange = _ref.onInputChange,
      validationMessage = _ref.validationMessage,
      props = _objectWithoutProperties(_ref, _excluded);

  var _useOptionFiltering = useOptionFiltering({
    value: value,
    options: options,
    onInputChange: onInputChange
  }),
      filteredOptions = _useOptionFiltering.filteredOptions,
      noOptionsLabel = _useOptionFiltering.noOptionsLabel,
      onFilter = _useOptionFiltering.onFilter;

  var handleChange = function handleChange(newValues) {
    onChange(newValues || []);
  };

  var placeholderProps = usePlaceholder(value, validationMessage);
  var resizeHappened = useRef();
  useEffect(function () {
    if (options.length && !resizeHappened.current) {
      window.dispatchEvent(new Event('resize'));
      resizeHappened.current = true;
    }
  }, [options.length]);
  return React.createElement(SelectMulti, _extends({}, props, placeholderProps, {
    removeOnBackspace: false,
    values: value,
    onChange: handleChange,
    isFilterable: true,
    onFilter: onFilter,
    options: filteredOptions,
    noOptionsLabel: noOptionsLabel,
    width: 410,
    maxWidth: "100%",
    maxHeight: 145,
    listLayout: {
      maxWidth: ['95vw', '90vw', '80vw', '65vw', '50vw'],
      width: 'auto'
    },
    noErrorIcon: true,
    validationType: validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type
  }));
};

export var TagList = styled(TagListComponent).withConfig({
  displayName: "TagList",
  componentId: "sc-s3oxgm-0"
})(["", "{", "}"], InputText, tokenStylePlaceholder);
