// @looker/filter-components/Filter/components/AdvancedFilter/components/DateFilter/components/YearMonth/YearMonth.js
import "core-js/modules/es.number.parse-int.js";
import "core-js/modules/es.number.constructor.js";
import "core-js/modules/es.array.index-of.js";
import { getMonths } from '@looker/filter-expressions';
import padStart from 'lodash/padStart';
import React from 'react';
import { useTranslation } from '../../../../../../../utils';
import { createOptions } from '../../../../../../utils/option_utils';
import { GroupSelect } from '../../../GroupSelect';
import { GroupInput } from '../../../GroupInput';
export var YearMonth = function YearMonth(_ref) {
  var _ref$item = _ref.item,
      id = _ref$item.id,
      month = _ref$item.month,
      year = _ref$item.year,
      onChange = _ref.onChange;

  var _useTranslation = useTranslation('get_months'),
      t = _useTranslation.t;

  var months = getMonths(t);
  var monthNumber = Number.parseInt(month, 10) - 1;
  var monthString = months[monthNumber];

  var yearChange = function yearChange(e) {
    onChange(id, {
      year: e.target.value
    });
  };

  var monthChange = function monthChange(value) {
    var monthValue = padStart(String(months.indexOf(value) + 1), 2, '0');
    onChange(id, {
      month: monthValue
    });
  };

  return React.createElement(React.Fragment, null, React.createElement(GroupSelect, {
    value: monthString,
    options: createOptions(months),
    onChange: monthChange,
    placement: "middle"
  }), React.createElement(GroupInput, {
    onChange: yearChange,
    value: year,
    placement: "right"
  }));
};
