// @looker/filter-components/Filter/components/AdvancedFilter/components/AddRemoveButtons/AddRemoveButtons.js
import { IconButton } from '@looker/components';
import { Add } from '@styled-icons/material/Add';
import { Close } from '@styled-icons/material/Close';
import { useTranslation } from '../../../../../utils';
import React from 'react';
export var AddRemoveButtons = function AddRemoveButtons(_ref) {
  var onAdd = _ref.onAdd,
      onRemove = _ref.onRemove,
      showAdd = _ref.showAdd,
      showRemove = _ref.showRemove;

  var _useTranslation = useTranslation('AddRemoveButtons'),
      t = _useTranslation.t;

  return React.createElement(React.Fragment, null, showRemove && React.createElement(IconButton, {
    icon: React.createElement(Close, null),
    size: "small",
    ml: "xsmall",
    label: t('Remove'),
    outline: false,
    onClick: onRemove,
    style: {
      marginTop: '2px'
    }
  }), showAdd && React.createElement(IconButton, {
    icon: React.createElement(Add, null),
    size: "small",
    ml: "xsmall",
    label: t('Add'),
    outline: false,
    onClick: onAdd,
    style: {
      marginTop: '2px'
    }
  }));
};
