// @looker/filter-components/Filter/components/AdvancedFilter/components/DateFilter/components/Relative/Relative.js
import { Box2 } from '@looker/components';
import React from 'react';
import { useTranslation } from '../../../../../../../utils';
import { GroupSelect } from '../../../GroupSelect';
import { MidInputLabel } from '../../../MidInputLabel';
import { Interval } from '../Interval';
export var Relative = function Relative(_ref) {
  var _ref$item = _ref.item,
      id = _ref$item.id,
      intervalType = _ref$item.intervalType,
      startInterval = _ref$item.startInterval,
      endInterval = _ref$item.endInterval,
      onChange = _ref.onChange,
      field = _ref.field;

  var _useTranslation = useTranslation('Relative'),
      t = _useTranslation.t;

  var options = [{
    value: 'ago',
    label: t('ago')
  }, {
    value: 'from now',
    label: t('from now')
  }];

  var intervalTypeChange = function intervalTypeChange(value) {
    onChange(id, {
      intervalType: value
    });
  };

  var startChange = function startChange(interval) {
    onChange(id, {
      startInterval: interval
    });
  };

  var endChange = function endChange(interval) {
    onChange(id, {
      endInterval: interval
    });
  };

  return React.createElement(Box2, {
    display: "flex",
    bg: "field"
  }, React.createElement(Interval, {
    placement: "middle",
    item: startInterval,
    onChange: startChange,
    field: field
  }), React.createElement(GroupSelect, {
    value: intervalType,
    options: options,
    onChange: intervalTypeChange,
    placement: "middle"
  }), React.createElement(MidInputLabel, null, "FOR"), React.createElement(Interval, {
    item: endInterval,
    onChange: endChange,
    field: field
  }));
};
