// @looker/filter-components/Filter/components/AdvancedFilter/components/DateFilter/components/Interval/Interval.js
import "core-js/modules/es.number.constructor.js";
import React from 'react';
import { useDateUnits, useFiscalIntervalUnits } from '../../../../../../constants';
import { showFiscalUnits } from '../../../../utils/show_fiscal_units';
import { GroupSelect } from '../../../GroupSelect';
import { GroupInput } from '../../../GroupInput';
export var Interval = function Interval(_ref) {
  var _ref$placement = _ref.placement,
      placement = _ref$placement === void 0 ? 'right' : _ref$placement,
      _ref$item = _ref.item,
      unit = _ref$item.unit,
      value = _ref$item.value,
      onChange = _ref.onChange,
      field = _ref.field;

  var valueChange = function valueChange(e) {
    return onChange({
      value: Number(e.target.value),
      unit: unit
    });
  };

  var unitChange = function unitChange(unit) {
    return onChange({
      value: value,
      unit: unit
    });
  };

  var fiscalIntervalUnits = useFiscalIntervalUnits();
  var dateUnits = useDateUnits();
  var options = showFiscalUnits(field) ? fiscalIntervalUnits : dateUnits;
  return React.createElement(React.Fragment, null, React.createElement(GroupInput, {
    onChange: valueChange,
    value: value,
    placement: "middle"
  }), React.createElement(GroupSelect, {
    value: unit,
    options: options,
    onChange: unitChange,
    placement: placement
  }));
};
