// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/DateFilter/components/RelativeTimeframes/components/RelativeTimeframePresets/RelativeTimeframePresets.js
"use strict";

require("core-js/modules/es.symbol.description.js");

require("core-js/modules/es.array.iterator.js");

require("core-js/modules/es.array.slice.js");

require("core-js/modules/es.array.from.js");

require("core-js/modules/es.weak-map.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RelativeTimeframePresets = void 0;

require("core-js/modules/es.array.map.js");

require("core-js/modules/es.object.values.js");

var _components = require("@looker/components");

var _Check = require("@styled-icons/material/Check");

var _ExpandMore = require("@styled-icons/material-rounded/ExpandMore");

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _utils = require("../../../../../../../../../utils");

var _get_relative_timeframe_presets = require("../../utils/get_relative_timeframe_presets");

var _relative_timeframe_types = require("../../../../types/relative_timeframe_types");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) arr2[i] = arr[i]; return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

const RelativeTimeframePresets = props => {
  const _useTranslation = (0, _utils.useTranslation)('RelativeTimeframePresets'),
        t = _useTranslation.t;

  const _useState = (0, _react.useState)(false),
        _useState2 = _slicedToArray(_useState, 2),
        showMore = _useState2[0],
        setShowMore = _useState2[1];

  const handleClick = e => {
    setShowMore(true);
    e.preventDefault();
  };

  return _react.default.createElement(_components.MenuList, {
    density: -1,
    iconGutter: true
  }, showMore ? _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: _relative_timeframe_types.PresetTimeframesMostRecent,
    delay: "rapid"
  })), _react.default.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: _relative_timeframe_types.PresetTimeframesRecent,
    delay: "simple"
  })), _react.default.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: _relative_timeframe_types.PresetTimeframesLeastRecent,
    delay: "moderate"
  })), _react.default.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: _relative_timeframe_types.PresetTimeframesThis,
    delay: "complex"
  })), _react.default.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: _relative_timeframe_types.PresetTimeframesPrevious,
    delay: "intricate"
  }))) : _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(PresetTimeframeGroup, _extends({}, props, {
    presetTimeframe: _relative_timeframe_types.PresetTimeframes,
    delay: "none",
    duration: "none"
  })), _react.default.createElement(CustomMenuItem, {
    detail: _react.default.createElement(_components.Icon, {
      size: 28,
      color: "ui3",
      pr: "xsmall",
      icon: _react.default.createElement(_ExpandMore.ExpandMore, null)
    }),
    onClick: handleClick
  }, t('More'))));
};

exports.RelativeTimeframePresets = RelativeTimeframePresets;

const PresetTimeframeGroup = ({
  duration = 'simple',
  delay,
  onPresetChange,
  presetTimeframe,
  value
}) => {
  const presets = (0, _get_relative_timeframe_presets.useRelativeTimeframePresets)();

  const _useContext = (0, _react.useContext)(_components.DialogContext),
        closeModal = _useContext.closeModal;

  const handleOnClick = timeframe => () => {
    onPresetChange(timeframe);
    closeModal === null || closeModal === void 0 ? void 0 : closeModal();
  };

  return _react.default.createElement(_react.default.Fragment, null, Object.values(presetTimeframe).map((timeframe, index) => {
    const current = typeof value === 'string' && value === timeframe;
    return _react.default.createElement(CustomMenuItem, {
      icon: current ? _react.default.createElement(_Check.Check, null) : undefined,
      key: index,
      onClick: handleOnClick(timeframe),
      selected: current
    }, _react.default.createElement(_components.FadeIn, {
      duration: duration,
      delay: delay
    }, presets[timeframe]));
  }), _react.default.createElement(_components.MenuDivider, null));
};

const CustomMenuItem = (0, _styledComponents.default)(_components.MenuItem).withConfig({
  displayName: "RelativeTimeframePresets__CustomMenuItem",
  componentId: "sc-1lkgww6-0"
})(["& > button{&[aria-current='true']{background-color:", ";", "{color:", ";}}&:hover{background-color:", ";}}"], ({
  theme: {
    colors
  }
}) => colors.background, _components.Icon, ({
  theme: {
    colors
  }
}) => colors.key, ({
  theme: {
    colors
  }
}) => colors.keySubtle);
