// @looker/filter-components/Filter/components/AdvancedFilter/components/TierFilter/TierFilter.js
var _excluded = ["item", "filterType", "enumerations", "field", "userAttributes", "onChange", "showMatchesAdvanced", "validationMessage"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import "core-js/modules/es.array.map.js";
import "core-js/modules/es.array.some.js";
import "core-js/modules/es.array.includes.js";
import "core-js/modules/es.string.includes.js";
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.object.assign.js";

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import { convertOptionToType, convertTypeToOption } from '@looker/filter-expressions';
import React from 'react';
import { GroupSelect } from '../GroupSelect';
import { tierFilterTypeToFilter } from './utils/tier_filter_type_to_filter';
import { useParamFilterOptions, useTierFilterOptions, useFilterOptions } from '../../utils';
import { ItemLayout } from '../ItemLayout';
import { createEnumeration } from '../../../../utils';
export var TierFilter = function TierFilter(_ref) {
  var item = _ref.item,
      filterType = _ref.filterType,
      enumerations = _ref.enumerations,
      field = _ref.field,
      userAttributes = _ref.userAttributes,
      onChange = _ref.onChange,
      showMatchesAdvanced = _ref.showMatchesAdvanced,
      validationMessage = _ref.validationMessage,
      rest = _objectWithoutProperties(_ref, _excluded);

  var typeChange = function typeChange(value) {
    return onChange(item.id, convertOptionToType(String(value)));
  };

  var selectValue = convertTypeToOption(item);
  var isParamFilter = field ? field.category === 'parameter' : false;
  var unescapeEnumerationValues = (field === null || field === void 0 ? void 0 : field.has_allowed_values) && isParamFilter;
  var unescapedEnumerations = enumerations === null || enumerations === void 0 ? void 0 : enumerations.map(createEnumeration(unescapeEnumerationValues));
  var FilterComponent = tierFilterTypeToFilter(item.type, isParamFilter, rest.allowMultipleOptions);
  var isValueSet = item.value && item.value.length > 0;
  var isValueInEnumeration = unescapedEnumerations === null || unescapedEnumerations === void 0 ? void 0 : unescapedEnumerations.some(function (e) {
    var _item$value;

    return (_item$value = item.value) === null || _item$value === void 0 ? void 0 : _item$value.includes(e.value);
  });

  if (isParamFilter && unescapedEnumerations !== null && unescapedEnumerations !== void 0 && unescapedEnumerations.length && isValueSet && !isValueInEnumeration) {
    item.value = [String(unescapedEnumerations[0].value)];
  }

  var paramFilterOptions = useParamFilterOptions();
  var tierFilterOptions = useTierFilterOptions();
  var options = useFilterOptions(isParamFilter ? paramFilterOptions : tierFilterOptions, isParamFilter ? false : showMatchesAdvanced);
  return React.createElement(ItemLayout, _extends({
    item: item
  }, rest), React.createElement(GroupSelect, {
    value: selectValue,
    options: options,
    onChange: typeChange,
    validationType: validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type,
    placement: item.type === 'anyvalue' ? undefined : 'left'
  }), React.createElement(FilterComponent, {
    item: item,
    onChange: onChange,
    borderRadiusLeft: 0,
    disableCreate: true,
    validationMessage: validationMessage,
    enumerations: unescapedEnumerations,
    userAttributes: userAttributes,
    filterType: filterType,
    placement: "right"
  }));
};
