// @looker/filter-components/Filter/components/AdvancedFilter/components/StringFilter/utils/string_filter_type_to_filter.js
import defaultTo from 'lodash/defaultTo';
import { MatchesAdvanced } from '../../MatchesAdvanced';
import { UserAttributes } from '../../UserAttributes';
import { MultiStringInput } from '../components/MultiStringInput';
import { StringInput } from '../components/StringInput';

var Blank = function Blank() {
  return '';
};

var filterTypeToStringMap = {
  "null": Blank,
  contains: MultiStringInput,
  match: MultiStringInput,
  startsWith: MultiStringInput,
  endsWith: MultiStringInput,
  blank: Blank,
  user_attribute: UserAttributes
};
var filterTypeToStringMapSingleValue = {
  contains: StringInput,
  match: StringInput,
  startsWith: StringInput,
  endsWith: StringInput
};
export var stringFilterTypeToFilter = function stringFilterTypeToFilter(type, isParameterField, allowMultipleValues) {
  if ((!allowMultipleValues || isParameterField) && filterTypeToStringMapSingleValue[type]) {
    return filterTypeToStringMapSingleValue[type];
  }

  return defaultTo(filterTypeToStringMap[type], MatchesAdvanced);
};
