// @looker/filter-components/Filter/components/AdvancedFilter/components/NumberFilter/NumberFilter.js
import "core-js/modules/es.object.keys.js";
import "core-js/modules/es.array.index-of.js";
import "core-js/modules/es.symbol.js";
import "core-js/modules/es.object.define-property.js";
import "core-js/modules/es.array.filter.js";
import "core-js/modules/es.object.get-own-property-descriptor.js";
import "core-js/modules/es.array.for-each.js";
import "core-js/modules/web.dom-collections.for-each.js";
import "core-js/modules/es.object.get-own-property-descriptors.js";
import "core-js/modules/es.object.define-properties.js";
import "core-js/modules/es.object.assign.js";
var _excluded = ["item", "filterType", "onChange", "validationMessage", "userAttributes", "showMatchesAdvanced"];

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import { convertOptionToType, convertTypeToOption, sanitizeNumber } from '@looker/filter-expressions';
import React from 'react';
import { useTranslation } from '../../../../../utils';
import { ItemLayout } from '../ItemLayout';
import { GroupSelect } from '../GroupSelect';
import { numberFilterTypeToFilter } from './utils/number_filter_type_to_filter';
import { useNumberFilterOptions, useFilterOptions } from '../../utils';
export var NumberFilter = function NumberFilter(_ref) {
  var _rest$field, _item$value;

  var item = _ref.item,
      filterType = _ref.filterType,
      onChange = _ref.onChange,
      validationMessage = _ref.validationMessage,
      userAttributes = _ref.userAttributes,
      showMatchesAdvanced = _ref.showMatchesAdvanced,
      rest = _objectWithoutProperties(_ref, _excluded);

  var isParameter = !!((_rest$field = rest.field) !== null && _rest$field !== void 0 && _rest$field.parameter);
  var numberFilterOptions = useNumberFilterOptions(isParameter);
  var options = useFilterOptions(numberFilterOptions, !isParameter && showMatchesAdvanced);

  var typeChange = function typeChange(value) {
    return onChange(item.id, sanitizeNumber(_objectSpread(_objectSpread({}, item), convertOptionToType(String(value)))));
  };

  var FilterComponent = numberFilterTypeToFilter(item.type, !!rest.allowMultipleOptions, isParameter);
  var selectValue = convertTypeToOption(item);
  var validationText = validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.message;

  var _useTranslation = useTranslation('NumberFilter'),
      t = _useTranslation.t;

  var placeholder = validationText || (!(item !== null && item !== void 0 && (_item$value = item.value) !== null && _item$value !== void 0 && _item$value.length) || item.value.length === 0 ? t('any value') : '');
  return React.createElement(ItemLayout, _extends({
    item: item
  }, rest), React.createElement(GroupSelect, {
    value: selectValue,
    options: options,
    onChange: typeChange,
    validationType: validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type,
    placement: item.type === 'null' ? undefined : 'left'
  }), React.createElement(FilterComponent, {
    item: item,
    onChange: onChange,
    validationMessage: validationMessage,
    placement: "right",
    userAttributes: userAttributes,
    filterType: filterType,
    placeholder: placeholder
  }));
};
