// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/ItemLayout/ItemLayout.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ItemLayout = void 0;

var _components = require("@looker/components");

var _react = _interopRequireDefault(require("react"));

var _AddRemoveButtons = require("../AddRemoveButtons");

var _OperatorLabel = require("../OperatorLabel");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ItemLayout = ({
  children,
  item,
  onAdd,
  onRemove,
  showOperator,
  showAdd,
  showRemove
}) => {
  const handleAdd = () => onAdd(item);

  const handleRemove = () => onRemove(item.id);

  return _react.default.createElement(_components.Box2, {
    role: "group",
    display: "flex",
    alignItems: "center"
  }, showOperator && _react.default.createElement(_components.Box2, {
    width: 44,
    mr: "medium",
    textAlign: "right"
  }, showOperator === true && _react.default.createElement(_OperatorLabel.OperatorLabel, {
    value: item.is
  })), children, _react.default.createElement(_AddRemoveButtons.AddRemoveButtons, {
    showAdd: showAdd,
    showRemove: showRemove,
    onAdd: handleAdd,
    onRemove: handleRemove
  }));
};

exports.ItemLayout = ItemLayout;
