// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/DateFilter/components/RelativeTimeframes/RelativeTimeframes.js
"use strict";

require("core-js/modules/es.array.iterator.js");

require("core-js/modules/es.weak-map.js");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RelativeTimeframes = void 0;

var _components = require("@looker/components");

var _react = _interopRequireWildcard(require("react"));

var _utils = require("../../../../../../../utils");

var _relative_timeframe_to_string = require("../../utils/relative_timeframe_to_string");

var _RelativeTimeframePopoverContent = require("./components/RelativeTimeframePopoverContent");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const RelativeTimeframes = ({
  value,
  onChange
}) => {
  const internalPopoverInstanceRef = (0, _react.useRef)();

  const _useTranslation = (0, _utils.useTranslation)('RelativeTimeframe'),
        t = _useTranslation.t;

  const handlePresetChange = selected => {
    onChange(selected);
  };

  const handleCustomChange = range => {
    onChange(range);
  };

  const handleNav = () => {
    var _internalPopoverInsta;

    (_internalPopoverInsta = internalPopoverInstanceRef.current) === null || _internalPopoverInsta === void 0 ? void 0 : _internalPopoverInsta.update();
  };

  const popoverContent = _react.default.createElement(_RelativeTimeframePopoverContent.RelativeTimeframePopoverContent, {
    value: value,
    onNav: handleNav,
    onCustomChange: handleCustomChange,
    onPresetChange: handlePresetChange
  });

  const popoverProps = {
    content: popoverContent,
    pin: true,
    placement: 'bottom-start'
  };

  const _usePopover = (0, _components.usePopover)(popoverProps),
        popover = _usePopover.popover,
        open = _usePopover.open,
        ref = _usePopover.ref,
        popperInstanceRef = _usePopover.popperInstanceRef;

  internalPopoverInstanceRef.current = popperInstanceRef.current;
  const label = (0, _relative_timeframe_to_string.useRelativeTimeframeToString)(value);

  const overlayTrigger = _react.default.createElement(_components.ChipButton, {
    ref: ref,
    onClick: open
  }, label);

  return _react.default.createElement(_react.default.Fragment, null, _react.default.createElement(_components.Breakpoint, {
    show: "mobile"
  }, _react.default.createElement(_components.Dialog, {
    content: _react.default.createElement(_components.DialogLayout, {
      header: t('Choose a Timeframe')
    }, popoverContent)
  }, overlayTrigger)), _react.default.createElement(_components.Breakpoint, {
    show: ['tablet', undefined]
  }, popover, overlayTrigger));
};

exports.RelativeTimeframes = RelativeTimeframes;
