// @looker/filter-components/Filter/components/AdvancedFilter/components/LocationFilter/LocationFilter.js
import "core-js/modules/es.array.includes.js";
import { Box, Flex } from '@looker/components';
import React from 'react';
import { GroupSelect } from '../GroupSelect';
import { locationFilterTypeToFilter } from './utils/location_filter_type_to_filter';
import { useLocationFilterOptions, useFilterOptions } from '../../utils';
export var LocationFilter = function LocationFilter(_ref) {
  var item = _ref.item,
      filterType = _ref.filterType,
      onChange = _ref.onChange,
      userAttributes = _ref.userAttributes,
      validationMessage = _ref.validationMessage,
      showMatchesAdvanced = _ref.showMatchesAdvanced;
  var locationFilterOptions = useLocationFilterOptions();
  var options = useFilterOptions(locationFilterOptions, showMatchesAdvanced);

  var locationTypeChange = function locationTypeChange(value) {
    return onChange(item.id, {
      type: value
    });
  };

  var FilterComponent = locationFilterTypeToFilter(item.type);
  return React.createElement(Flex, {
    flexDirection: "row",
    alignItems: "center"
  }, React.createElement(Box, null, React.createElement(GroupSelect, {
    value: item.type,
    options: options,
    onChange: locationTypeChange,
    validationType: validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type,
    placement: ['null', 'notnull', 'anyvalue'].includes(item.type) ? undefined : 'left'
  })), React.createElement(FilterComponent, {
    item: item,
    onChange: onChange,
    validationMessage: validationMessage,
    userAttributes: userAttributes,
    filterType: filterType
  }));
};
