// @looker/filter-components/Filter/components/ControlFilter/components/ButtonGroup/ButtonGroup.js
import { ButtonGroup as ButtonGroupComponent, ButtonItem, ProgressCircular } from '@looker/components';
import React from 'react';
import styled, { css } from 'styled-components';
import { ERROR_TYPE } from '../../../../../constants';
export var ButtonGroup = styled(function (_ref) {
  var className = _ref.className,
      isLoading = _ref.isLoading,
      onChange = _ref.onChange,
      options = _ref.options,
      value = _ref.value;
  return isLoading ? React.createElement(ProgressCircular, {
    size: "medium"
  }) : React.createElement(ButtonGroupComponent, {
    className: className,
    onChange: onChange,
    options: options,
    value: value
  });
}).withConfig({
  displayName: "ButtonGroup",
  componentId: "sc-ikfypj-0"
})(["", ""], function (_ref2) {
  var validationMessage = _ref2.validationMessage;
  return (validationMessage === null || validationMessage === void 0 ? void 0 : validationMessage.type) === ERROR_TYPE && css(["", "{border-color:", ";}"], ButtonItem, function (_ref3) {
    var theme = _ref3.theme;
    return theme.colors.criticalBorder;
  });
});
