// @looker/filter-components/cjs/Filter/components/AdvancedFilter/components/LocationFilter/components/LocationExact/LocationExact.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LocationExact = void 0;

var _components = require("@looker/components");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _constants = require("../../../../../../constants");

var _GroupInput = require("../../../GroupInput");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const LocationExact = ({
  item,
  onChange,
  latString = 'LATITUDE',
  lonString = 'LONGITUDE',
  lat = 'lat',
  lon = 'lon',
  placement = 'right'
}) => {
  const latChange = event => {
    if (Number(event.currentTarget.value) <= _constants.MAX_LATITUDE && Number(event.currentTarget.value) >= _constants.MIN_LATITUDE) {
      onChange(item.id, {
        [lat]: event.currentTarget.value
      });
    }
  };

  const lonChange = event => {
    if (Number(event.currentTarget.value) <= _constants.MAX_LONGITUDE && Number(event.currentTarget.value) >= _constants.MIN_LONGITUDE) {
      onChange(item.id, {
        [lon]: event.currentTarget.value
      });
    }
  };

  return _react.default.createElement(_components.Flex, {
    flexDirection: "row"
  }, _react.default.createElement(_GroupInput.GroupInput, {
    before: _react.default.createElement(Prefix, null, latString),
    value: String(item.lat || ''),
    type: "number",
    onChange: latChange,
    placement: "middle",
    minWidth: `${latString.length / 2 + 5.5}em`
  }), _react.default.createElement(_GroupInput.GroupInput, {
    before: _react.default.createElement(Prefix, null, lonString),
    value: String(item.lon || ''),
    type: "number",
    onChange: lonChange,
    placement: placement,
    minWidth: `${lonString.length / 2 + 5.5}em`
  }));
};

exports.LocationExact = LocationExact;

const Prefix = _styledComponents.default.span.withConfig({
  displayName: "LocationExact__Prefix",
  componentId: "sc-cjs9jc-0"
})(["color:", ";font-size:", ";white-space:nowrap;"], ({
  theme: {
    colors
  }
}) => colors.text1, ({
  theme: {
    fontSizes
  }
}) => fontSizes.xsmall);
